bin/transd.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python
+
+import argparse
+import jsonschema
+import logging
+import os
+import queue
+import threading
+import time
+import yaml
+import dbb_buffer_mngr as mgr
+
+
+logger = logging.getLogger("dbb_buffer_mngr")
+
+
+def parse_args():
+    """Parse command line arguments.
+
+    Returns
+    -------
+    argparse.Namespace
+        A namespace populated with arguments and their values.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-c", "--config", type=str, default=None,
+                        help="configuration file in YAML format")
+    parser.add_argument("-v", "--validate", action="store_true", default=False,
+                        help="validate configuration file")
+    return parser.parse_args()
+
+
+def set_logger(options=None):
+    """Configure logger.
+
+    Parameters
+    ----------
+    options : dict, optional
+       Logger settings. If None (default), default settings will be used.
+    """
+    settings = {
```

Comment:
I'm slightly confused here. What do you mean by:

> Since the dict isn't used as a dict anywhere

If the dictionary `options` is provided, its key/value pairs will be used to change logger settings from their default values to ones specified by user (lines 46-47).

Regarding using `options.get()` approach, it implies that `options` is always a dictionary. It is not the case in the current implementation. If there are no logging related entries in application configuration file, `options` will be set to **None**.

Finally, I didn't use optional parameters in `set_logger()` as I'm not really sure at this stage how many aspect of logging we want to make configurable. If we end up with just a few (say 3), I have no strong objections of making them optional arguments of the `set_logger()`. However, if there will be more I would prefer to stick to having it just one argument encapsulating its entire configuration.

Reply:
By using it as a dict, I was referring to the fact that other than using it for the update call, its only use is to access individual values (i.e., the dict isn't passed to any other function).  The options=None case is easy to remedy by a check early in the function and setting it to an empty dict if None.   I didn't specify that it had to be a code change but instead could just add a comment.